python/mdvtools/tests/test_stress_performance.py

```
@@ -0,0 +1,455 @@
+        start_memory = get_memory_usage()
+        
+        adata = factory.create_large(20000, 3000)
+        
+        creation_time = time.time() - start_time
+        creation_memory = get_memory_usage()
+        
+        print(f"Large dataset creation: {creation_time:.2f}s, Memory: {creation_memory['rss']:.1f}MB")
+        
+        # Test conversion
+        with temp_mdv_project() as test_dir:
+            with suppress_anndata_warnings():
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)